api/pkg/s3/utils.go

```diff
@@ -0,0 +1,143 @@
+/*
+ * Minio Cloud Storage, (C) 2015 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package s3
+
+import (
+	"encoding/base64"
+	"encoding/xml"
+	"errors"
+	"io"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+// xmlDecoder provide decoded value in xml.
+func xmlDecoder(body io.Reader, v interface{}) error {
+	d := xml.NewDecoder(body)
+	return d.Decode(v)
+}
+
+// checkValidMD5 - verify if valid md5, returns md5 in bytes.
+func checkValidMD5(md5 string) ([]byte, error) {
+	return base64.StdEncoding.DecodeString(strings.TrimSpace(md5))
+}
+
+/// http://docs.aws.amazon.com/AmazonS3/latest/dev/UploadingObjects.html
+const (
+	// maximum object size per PUT request is 5GiB
+	maxObjectSize = 1024 * 1024 * 1024 * 5
+	// minimum Part size for multipart upload is 5MB
+	minPartSize = 1024 * 1024 * 5
+	// maximum Part ID for multipart upload is 10000 (Acceptable values range from 1 to 10000 inclusive)
+	maxPartID = 10000
+)
+
+// isMaxObjectSize - verify if max object size
+func isMaxObjectSize(size int64) bool {
+	return size > maxObjectSize
+}
+
+// Check if part size is more than or equal to minimum allowed size.
+func isMinAllowedPartSize(size int64) bool {
+	return size >= minPartSize
+}
+
+// isMaxPartNumber - Check if part ID is greater than the maximum allowed ID.
+func isMaxPartID(partID int) bool {
+	return partID > maxPartID
+}
+
+func contains(stringList []string, element string) bool {
+	for _, e := range stringList {
+		if e == element {
+			return true
+		}
+	}
+	return false
+}
+
+/*
```

Comment:
Please remove unused codes.

Reply:
ok, the unused codes will be removed, thanks.